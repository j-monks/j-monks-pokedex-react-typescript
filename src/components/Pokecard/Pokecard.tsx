import './Pokecard.css'

interface PokecardProps {
    spriteUrl?: string,
    name: string,
    onClickPokemon: (pokemonName: string) => void
}

 const Pokecard = ({spriteUrl, name, onClickPokemon}: PokecardProps) => {
    return (
        <div data-testid={`pokecard-${name}`} onClick={() => onClickPokemon(name)} className="pokecard">
            <img className="pokemon-sprite" src={spriteUrl} alt="pokemon" />
            <p>{ name }</p>
        </div>
    );
}

export default Pokecard;