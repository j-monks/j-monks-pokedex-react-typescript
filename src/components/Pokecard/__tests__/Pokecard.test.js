import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import Pokecard from '../Pokecard'

afterEach(() => {
    cleanup()
})

test('should render pokecard component', () => {
    const pokecard = {
        spriteUrl: 'http://img.pokemondb.net/sprites/black-white/normal/bulbasaur.png',
        name: 'bulbasaur'
    }
    render(<Pokecard name={pokecard.name} spriteUrl={pokecard.spriteUrl}/>)
    const pokecardElement = screen.getByTestId('pokecard-bulbasaur')
    expect(pokecardElement).toBeInTheDocument()
    expect(pokecardElement).toHaveTextContent('bulbasaur')
})

test('matches snapshot', () => {
    const pokecard = {
        spriteUrl: 'http://img.pokemondb.net/sprites/black-white/normal/bulbasaur.png',
        name: 'bulbasaur'
    }
    const tree = renderer.create(<Pokecard name={pokecard.name} spriteUrl={pokecard.spriteUrl}/>).toJSON()
    expect(tree).toMatchSnapshot()
})