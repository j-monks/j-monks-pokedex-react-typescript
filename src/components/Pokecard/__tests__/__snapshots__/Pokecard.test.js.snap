// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="pokecard"
  data-testid="pokecard-bulbasaur"
  onClick={[Function]}
>
  <img
    alt="pokemon"
    className="pokemon-sprite"
    src="http://img.pokemondb.net/sprites/black-white/normal/bulbasaur.png"
  />
  <p>
    bulbasaur
  </p>
</div>
`;
